src/drawing/handlers.ts

```diff
@@ -24,6 +24,13 @@ import { BoardPage } from "./page"
 export function handleSetTool(tool: Partial<Tool>): void {
     drawing.setTool(tool)
     board.clearTransform()
+
+    const { activeTextfield } = board.getState()
+    if (activeTextfield?.textfield?.text) {
+        // Add back stroke which is being edited before clearing
+        handleAddStrokes([activeTextfield], false)
+    }
+    board.clearActiveTextfield()
```

Comment:
I'm not a fan of this logic being here in the handlers. Can we move it somewhere else, maybe the state?

Reply:
You can't call `handleAddStrokes` from inside of a state, since the handlers import the states. I think it's in the right place honestly, i'll separate this logic in another function to keep the handleSetTool clean ok?